src/js/views/globals/add-patient/add-patient_views.js

```diff
@@ -0,0 +1,325 @@
+import { isEmpty, keys, omit } from 'underscore';
+import Backbone from 'backbone';
+import hbs from 'handlebars-inline-precompile';
+import { View } from 'marionette';
+import { mixinState } from 'marionette.toolkit';
+
+import Datepicker from 'js/components/datepicker';
+import Droplist from 'js/components/droplist';
+import GroupsManagerComponent from 'js/components/groups-manager';
+
+import InputWatcherBehavior from 'js/behaviors/input-watcher';
+
+import intl from 'js/i18n';
+import trim from 'js/utils/formatting/trim';
+
+import AddPatientTemplate from './add-patient.hbs';
+
+import './add-patient.scss';
+
+const i18n = intl.globals.addPatient.addPatientViews;
+
+const InputView = View.extend({
+  template: hbs`<input class="input-primary w-100 js-input {{#if hasError}}has-error{{/if}}" value="{{ value }}" />`,
+  templateContext() {
+    const errors = this.getOption('state').get('errors');
+
+    return {
+      hasError: errors && errors[this.getOption('errorField')],
+      value: this.model.get(this.getOption('attr')),
+    };
+  },
+  behaviors: [InputWatcherBehavior],
+  ui: {
+    input: '.js-input',
+  },
+  initialize({ state, errorField }) {
+    this.listenTo(state, 'change:errors', () => {
```

Comment:
Moving this functionality to the parent view causes focus to be hijacked when a user is correcting an error. For example: They submit and get the `similar patient exists` error, then they try to modify the last name. When the error is cleared after the first letter they type, the whole form rerenders and changes the focus to the beginning of the first name input.